beacon/engine/types.go

```diff
@@ -297,10 +297,19 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 		withdrawalsRoot = &h
 	}
 
+	isthmusEnabled := bType.IsIsthmus(data.Timestamp)
 	var requestsHash *common.Hash
 	if requests != nil {
+		if !isthmusEnabled {
+			return nil, fmt.Errorf("requests should be nil for pre-Isthmus blocks")
+		}
```

Comment:
This seems like a new unintended branch saying "if requests are not nil, pre-isthmus, error". This didn't error before so why are we introducing this?

Reply:
good point - yeah this should be validated in the new payload code already, so no reason to add this here. I think we probably should still have the `isthmusEnabled` condition in the below if statement since it's only a requirement for Isthmus, not Prague. Not sure how important it is for `op-geth` to be able to run L1 still, but might as well maintain as much compatibility as possible.

Reply:
resolved here: eee1602bca4f55693fe82b97df12d91c79edeaab